containers/Dockerfile.base-spack

```diff
@@ -59,37 +59,31 @@ ARG SPACK_REPO_VERSION=releases/${SPACK_VERSION}
 ARG SPACK_PACKAGES_REPO_VERSION=main
 ARG SPACK_CONFIG_REPO_VERSION=main
 ARG SPACK_ARCH=linux-rocky8-x86_64
+ARG SPACK_CONFIG_DIR=/opt/spack-config/${SPACK_VERSION}/ci
 ARG COMPILER_PKG_NAME=intel-oneapi-compilers
 ARG COMPILER_PKG_VERSION=2021.2.0
 ARG COMPILER_NAME=intel
 ARG COMPILER_VERSION=2021.2.0
 
 ENV SPACK_ROOT=/opt/spack
 ENV GNUPGHOME=${SPACK_ROOT}/opt/spack/gpg
-ENV ENV_SPACK_ARCH=${SPACK_ARCH}
-ENV SPACK_PACKAGES_REPO_ROOT=/opt/spack-packages
-ENV SPACK_CONFIG_REPO_ROOT=/opt/spack-config
-ENV SPACK_CONFIG_DIR=${SPACK_CONFIG_REPO_ROOT}/${SPACK_VERSION}/ci
-ENV ENV_COMPILER_PKG_NAME=${COMPILER_PKG_NAME}
-ENV ENV_COMPILER_PKG_VERSION=${COMPILER_PKG_VERSION}
-ENV ENV_COMPILER_NAME=${COMPILER_NAME}
-ENV ENV_COMPILER_VERSION=${COMPILER_VERSION}
 
 LABEL au.org.access-nri.image.spack-repo-version ${SPACK_REPO_VERSION}
 LABEL au.org.access-nri.image.spack-packages-repo-version ${SPACK_PACKAGES_REPO_VERSION}
-LABEL au.org.access-nri.image.compiler ${ENV_COMPILER_NAME}@${ENV_COMPILER_VERSION}
-LABEL au.org.access-nri.image.arch ${ENV_SPACK_ARCH}
+LABEL au.org.access-nri.image.spack-config-repo-version ${SPACK_CONFIG_REPO_VERSION}
+LABEL au.org.access-nri.image.compiler ${COMPILER_NAME}@${COMPILER_VERSION}
+LABEL au.org.access-nri.image.arch ${SPACK_ARCH}
 
 SHELL ["/bin/bash", "-c"]
 
 # Install spack
 RUN git clone -c feature.manyFiles=true ${SPACK_GIT_URL} ${SPACK_ROOT} --branch ${SPACK_REPO_VERSION} --single-branch --depth=1
 
 # Install ACCESS-NRI's spack-packages repo
-RUN git clone  https://github.com/ACCESS-NRI/spack-packages.git ${SPACK_PACKAGES_REPO_ROOT} --branch ${SPACK_PACKAGES_REPO_VERSION}
+RUN git clone  https://github.com/ACCESS-NRI/spack-packages.git /opt/spack-packages --branch ${SPACK_PACKAGES_REPO_VERSION}
```

Comment:
Why do we need to remove the `SPACK_PACKAGES_REPO_ROOT` (and `SPACK_CONFIG_REPO_ROOT`)? Couldn't it just be converted into an `ARG` rather than an `ENV`, if we don't want the variable after the build?
It might be unlikely to have a `spack-{packages,config}` somewhere other than `$SPACK_ROOT/..`, but it may happen one day, and we might like to have that customisation not hard-coded in the Dockerfile. What do you reckon?

Reply:
`/opt/spack-packages` can't be changed, it's also hardcoded in `spack-config` repo's `repos.yaml`.